src/Microsoft.Android.Build.BaseTasks/XDocumentExtensions.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml.XPath;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.Android.Build.Tasks
+{
+	public static class XDocumentExtensions
+	{
+		const string PathsElementName = "Paths";
+
+		public static ITaskItem[] GetPathsAsTaskItems (this XDocument doc, params string[] paths)
+		{
+			var e = doc.Elements (PathsElementName);
+			foreach (var p in paths)
+				e = e.Elements (p);
+			return e.Select (ToTaskItem).ToArray ();
+		}
+
+		static ITaskItem ToTaskItem (XElement element)
+		{
+			var taskItem = new TaskItem (element.Value);
+			foreach (var attribute in element.Attributes ()) {
+				taskItem.SetMetadata (attribute.Name.LocalName, attribute.Value);
+			}
+			return taskItem;
+		}
+
+		public static IEnumerable<XElement> ToXElements (ICollection<ITaskItem> items, string itemName, string[] knownMetadata)
```

Comment:
This isn't an extension method; should it be placed into a separate type?  Or made non-`public`?

Reply:
@pjcollins can we change this one to an extension method?

So the usage changes to:
```diff
--XDocumentExtensions.ToXElements (ResolvedResourceDirectories, "ResolvedResourceDirectory", knownMetadata)
++ResolvedResourceDirectories.ToXElements ("ResolvedResourceDirectory", knownMetadata)
```

Reply:
I moved this to an `ITaskItemExtensions` class. However, since this is not used by monodroid, we could just remove it and put a new `ITaskItemExtensions.cs` in `xamarin/xamarin-android` instead?

Reply:
If monodroid doesn't use it, we can keep it in xamarin-android.

Reply:
I removed it, we can make this change in xamarin-android instead.